crates/payload/builder/src/payload.rs

```diff
@@ -151,7 +184,11 @@ impl PayloadBuilderAttributes {
 /// Generates the payload id for the configured payload
 ///
 /// Returns an 8-byte identifier by hashing the payload components with sha256 hash.
-pub(crate) fn payload_id(parent: &H256, attributes: &PayloadAttributes) -> PayloadId {
+pub(crate) fn payload_id(
+    parent: &H256,
+    attributes: &PayloadAttributes,
+    #[cfg(feature = "optimism")] txs: &Vec<TransactionSigned>,
```

Comment:
Not sure of best practices here, but it feels to me like public apis should generally not change on feature enabling.

Can we change this to concretely accept a vec of signed transactions? Or alternatively create a new method that wraps + refactors some internal logic?

Reply:
pub(crate), so it should be ok for now - once upstreamed will be maintained so less breakage than introducing a new function that also has to be maintained.